l10n_it_reverse_charge/tests/rc_common.py

```diff
@@ -186,7 +186,7 @@ def _create_rc_types(cls):
                 "method": "selfinvoice",
                 "partner_type": "other",
                 "with_supplier_self_invoice": True,
-                "partner_id": cls.env.ref("base.main_partner").id,
+                "partner_id": cls.env.company.partner_id.id,
```

Comment:
Hai modo di riportare l'errore che viene sollevato senza questa modifica?

Reply:
La situazione qui è leggermente complicata e fa riferimento a https://github.com/OCA/multi-company/pull/526.

Sostanzialmente prima veniva sempre e in ogni caso preso in considerazione come invoice tra diverse company, anche se fatto a se stesso, e senza prodotti nelle line, un invoice tra diverse company genera errore.

Attraverso la PR linkata prima il problema in origine è stato sistemato, siccome un invoice fatto a se stessi non dovrebbe essere considerato come invoice tra diverse company, ma comunque `base.main_partner` non faceva parte della stessa company al momento della creazione, per cui è meglio utilizzare quella salvata nell'`env`.

Lascio comunque il trace dell'errore qui sotto:

```
2023-11-23 11:27:49,679 1 ERROR devel odoo.addons.l10n_it_reverse_charge.tests.test_rc: ERROR: TestReverseCharge.test_supplier_extraEU_no_outstanding_payment
Traceback (most recent call last):
  File "/opt/odoo/auto/addons/l10n_it_reverse_charge/tests/test_rc.py", line 162, in test_supplier_extraEU_no_outstanding_payment
    invoice = self.create_invoice(
  File "/opt/odoo/auto/addons/l10n_it_reverse_charge/tests/rc_common.py", line 93, in create_invoice
    invoice = cls.init_invoice(
  File "/opt/odoo/auto/addons/account/tests/common.py", line 396, in init_invoice
    rslt.action_post()
  File "/opt/odoo/auto/addons/sale/models/account_move.py", line 14, in action_post
    res = super(AccountMove, self).action_post()
  File "/opt/odoo/auto/addons/l10n_it_reverse_charge/models/account_move.py", line 475, in action_post
    invoice.generate_supplier_self_invoice()
  File "/opt/odoo/auto/addons/l10n_it_reverse_charge/models/account_move.py", line 460, in generate_supplier_self_invoice
    supplier_invoice.action_post()
  File "/opt/odoo/auto/addons/sale/models/account_move.py", line 14, in action_post
    res = super(AccountMove, self).action_post()
  File "/opt/odoo/auto/addons/l10n_it_reverse_charge/models/account_move.py", line 464, in action_post
    ret = super().action_post()
  File "/opt/odoo/auto/addons/account/models/account_move.py", line 2823, in action_post
    self._post(soft=False)
  File "/opt/odoo/auto/addons/sale/models/account_invoice.py", line 75, in _post
    posted = super()._post(soft)
  File "/opt/odoo/auto/addons/purchase_stock/models/account_invoice.py", line 173, in _post
    return super()._post(soft)
  File "/opt/odoo/auto/addons/stock_account/models/account_move.py", line 53, in _post
    posted = super()._post(soft)
  File "/opt/odoo/auto/addons/account_invoice_inter_company/models/account_move.py", line 74, in _post
    self.create_counterpart_invoices()
  File "/opt/odoo/auto/addons/account_invoice_inter_company/models/account_move.py", line 91, in create_counterpart_invoices
    src_invoice.with_company(dest_company.id).with_context(
  File "/opt/odoo/auto/addons/purchase_sale_inter_company/models/account_move.py", line 13, in _inter_company_create_invoice
    res = super()._inter_company_create_invoice(dest_company)
  File "/opt/odoo/auto/addons/account_invoice_inter_company/models/account_move.py", line 173, in _inter_company_create_invoice
    raise UserError(
odoo.exceptions.UserError: The invoice line 'test line' doesn't have a product. All invoice lines should have a product for inter-company invoices.
```

Reply:
Grazie della spiegazione.